### DISCUSSION TITLE
Running mHM at finer resolution

### DISCUSSION THREAD

**Discussion by bibek-ux** [mhm-ufz/mHM#101] [Q&A]
Hello everyone,
                 I am student from Nepal and quite new to mHM. I have ran this model at 0.25 resolution( hydrological and routing) but running at finer resolution produces an ERROR message: 

Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

Backtrace for this error:
#0  0x7f6ea002351f in ???
#1  0x7f6ea049d9ea in ???
#2  0x7f6ea04b8503 in ???
#3  0x7f6ea044ca73 in ???
#4  0x7f6ea04457f5 in ???
#5  0x7f6ea049ac1d in ???
#6  0x7f6ea0547f19 in ???
#7  0x7f6ea04e5c55 in ???
#8  0x7f6ea0445e14 in ???
#9  0x5569499fe09a in ???
#10  0x7f6ea000ad8f in ???
#11  0x7f6ea000ae3f in ???
#12  0x5569499fe0e0 in ???
                                I have provided latlon at the corresponding resolution as well. 
                                Any help regarding this would be much appreciated. Thank you.

Best regards,
Bibek Pathak

**Comment by pallavhydro** [comment]
Hi Bibek

Welcome to the mHM world.

Could you do the following:
1. finer resolution = ? (or just share your `mhm.nml`) Usually it's best to work with the following relationship between the $L_1$ (hydrology), $L_{11}$ (routing), and $L_0$ (morphology) resolutions


$L_1\ (or\ L_{11}) =  2^n\ L_0$


where, n = 1, 2, ...

L0 resolution is the resolution of the morphology (e.g. dem.asc). For instance, for a $L_0$ resolution of, say, 1/512 $^\circ$, valid resolutions are 0.5 $^\circ$, 0.25 $^\circ$, 0.125 $^\circ$, .... 

2. If you have the know how, please `compile mhm in debug mode` and re-run to see if that gives out more error details during the segfault. The error message you are getting atm hardly has anything to backtrace.

Cheers
Pallav

**Reply by bibek-ux** [reply]
Hi pallav dai, 
        Applozies for my late response. I think I have  figured out what was causing the above issue. I was trying had set the routing and hydrological resolution at 0.0625 degrees and choose processcase(5) = -1 ( LAI corrected PET). The LAI data provided was at the resolution of 0.25. Keeping the processcase(5)= 0 (aspect corrected PET) ran the model without the above error.  

     I have another doubt that I wanted to ask you dai. 

     I have a nc file with where lat lon values of PET are stored in 2D array instead of 1D unlike pre.nc or tavg .nc both of which have lat, lon stored as 1d array. Can you help me understand if mhm model is capable of handling PET files like this. 


Best,
Bibek Pathak

![Screenshot 2024-10-04 172344](https://github.com/user-attachments/assets/650fd83a-052e-4c6f-bbaa-9ee54b7f8ca0)

![image](https://github.com/user-attachments/assets/f0e89fe6-7c13-4c96-848f-9c6a3b0c61d1)

**Reply by StephanThober** [reply]
Hi Bibek,

This should work fine. However, be sure that the two coordinate systems you have (the ones for pre.nc, tavg.nc and the one for pet.nc) do align. These two should match the one specified in your latlon.nc file!

I just made the following test. I removed the lat lon variables from the pre.nc and tavg.nc file in the test domain using nco:
```
mv pre.nc pre_original.nc
ncks -Cx -v lat,lon pre_original.nc pre.nc
```
mhm worked just fine!

I hope this helps!

Best,
Stephan

**Reply by pallavhydro** [reply]
Hi @bibek-ux 

If the discussion here solved your issue, I'd appreciate you mark the response that solved the issue.

Thanks in advance.
Pallav
